client/lib/domains/index.js

```diff
@@ -87,6 +87,10 @@ function canRedirect( siteId, domainName, onComplete ) {
 		return;
 	}
 
+	if ( ! domainName.match(/^(http)/) ) {
```

Comment:
First of all, thank you for your contribution, @VagishVela! :bow:

I think this regex is not sufficient - in the rare (but possible) case that the domain starts with `http`, it will not work :) So, for example, someone has a `httprocks.com` domain - this code will not add the `http://` prefix in that case. I think something more comprehensive, like `^https?://` is needed (BTW, no need for parenthesis around `http`).

BTW, have you checked what happens if you try to add a `https://` redirect here? :) Your regex will pass it - but I've got a feeling that the backend will return an error. It'd be great if you could handle that case too (but I understand if you prefer to fix one case at a time ;)).